### PR TITLE
Change listener to visitor in antlr in order to improve prase speed.

### PR DESCRIPTION
**Pull request by Genius-pig**
# benchmark Performance
performance |visitor | listener
---|---|---
throughput (point/s) | 2061421.49  | 1635560.00
time (second) | 29.11  | 36.68

# config

DB_SWITCH: IoTDB
OPERATION_PROPORTION: 1:0:0:0:0:0:0:0:0
ENABLE_THRIFT_COMPRESSION: false
INSERT_DATATYPE_PROPORTION: 1:1:1:1:1:1
ENCODING(BOOLEAN/INT32/INT64/FLOAT/DOUBLE/TEXT): PLAIN/PLAIN/PLAIN/PLAIN/PLAIN/PLAIN
IS_CLIENT_BIND: true
CLIENT_NUMBER: 20
GROUP_NUMBER: 20
DEVICE_NUMBER: 20
SENSOR_NUMBER: 300
BATCH_SIZE: 10
LOOP: 1000
POINT_STEP: 5000
QUERY_INTERVAL: 250000
IS_OVERFLOW: false
OVERFLOW_MODE: 0
OVERFLOW_RATIO: 0.5

# basal principle
Reduced syntax tree traversal

# performance (only include parse part)
308a1facdc930cd487e27ed836e9d0a

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/qp/sql/IoTDBSqlVisitor.java [hunk 1]
```diff
@@ -0,0 +1,1739 @@
...
+        parseDisableAlign(queryOp);
+      }
+    }
+    return queryOp;
+  }
+
+  @Override
+  public Operator visitAlignByDeviceStatementOrDisableAlignInSpecialClause(
+      AlignByDeviceStatementOrDisableAlignInSpecialClauseContext ctx) {
+    if(ctx.alignByDeviceClauseOrDisableAlign() != null) {
```
Comment (Ring-k) [inline server/src/main/java/org/apache/iotdb/db/qp/sql/IoTDBSqlVisitor.java:940]: Why is this necessary?

**Review by Ring-k — CHANGES_REQUESTED** [review]
Good Job! For system softwares, minor improvement counts.

File: server/src/main/java/org/apache/iotdb/db/qp/sql/IoTDBSqlVisitor.java [hunk 2]
```diff
@@ -0,0 +1,1739 @@
...
+    return alterTimeSeriesOperator;
+  }
+
+  @Override
+  public Operator visitInsertStatement(InsertStatementContext ctx) {
+    InsertOperator insertOp = new InsertOperator(SQLConstant.TOK_INSERT);
+    SelectOperator selectOp = new SelectOperator(SQLConstant.TOK_SELECT);
+    selectOp.addSelectPath(parsePrefixPath(ctx.prefixPath()));
+    insertOp.setSelectOperator(selectOp);
+    parseInsertColumnSpec(ctx.insertColumnSpec(), insertOp);
```
[the remark this answers is not in the document]
Comment (Genius-pig) [reply server/src/main/java/org/apache/iotdb/db/qp/sql/IoTDBSqlVisitor.java]: fixed

File: server/src/main/java/org/apache/iotdb/db/qp/sql/IoTDBSqlVisitor.java (hunk 1 above)
Comment (Genius-pig) [reply server/src/main/java/org/apache/iotdb/db/qp/sql/IoTDBSqlVisitor.java]: fixed

File: server/src/main/java/org/apache/iotdb/db/qp/logical/crud/QueryOperator.java [hunk 3]
```diff
@@ -56,11 +57,31 @@
 
   private boolean ascending = true;
 
+  private Map<String, Object> props;
```
Comment (kr11) [inline server/src/main/java/org/apache/iotdb/db/qp/logical/crud/QueryOperator.java:60]: "props" is a little general and may be confused if the original query involves other parameters/ properties, shall we rename it to "indexProps"?

File: server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java [hunk 4]
```diff
@@ -608,9 +604,11 @@ private PhysicalPlan transformQuery(QueryOperator queryOperator, int fetchSize)
         }
       }
     }
-    if(queryOperator instanceof QueryIndexOperator) {
-      ((QueryIndexPlan) queryPlan).setIndexType(((QueryIndexOperator) queryOperator).getIndexType());
-      ((QueryIndexPlan) queryPlan).setProps(((QueryIndexOperator) queryOperator).getProps());
+    if(queryOperator.getIndexType() != null) {
+      if(queryPlan instanceof QueryIndexPlan) {
```
Comment (kr11) [inline server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java:608]: As initialized at line 397, will `getIndexType() != null` and `instanceof QueryIdnexPlan` be equivalent?

File: server/src/main/java/org/apache/iotdb/db/qp/logical/crud/QueryOperator.java (hunk 3 above)
Comment (Genius-pig) [reply server/src/main/java/org/apache/iotdb/db/qp/logical/crud/QueryOperator.java:60]: yes, I think it's ok

**Review by kr11 — APPROVED** [review]
Great works!

File: server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java (hunk 4 above)
Comment (Genius-pig) [reply server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java:608]: I think it's equivalent

**Review by Ring-k — APPROVED** [review]
